server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -117,15 +117,22 @@ private void consumeWorkqueue() {
         String json = new String(delivery.getBody(), StandardCharsets.UTF_8);
 
         TaskId taskId = TaskId.fromString(delivery.getProperties().getHeaders().get(TASK_ID).toString());
-
+        Task task;
         try {
-            Task task = taskSerializer.deserialize(json);
+            task = taskSerializer.deserialize(json);
+        } catch (Exception e) {
+            LOGGER.warn("Unable to deserialize submitted Task  {}", taskId.asString(), e);
+            worker.fail(taskId, Optional.empty(), e);
+            return Mono.empty();
+        } finally {
             delivery.ack();
```

Comment:
you put `ack()` inside the `finally` block, that makes me thing you want to ack no matter what happened, so, why not putting .ack() at the very first line of `executeTask()`. I think it helps to respond to rabbitmq a bit faster